### PR TITLE
Header toolbar unit testing

### PR DESCRIPTION
**Pull request by nunibaranes**
Screen Shot 2020-10-12 at 12 07 36

Adds tests to the following components:
`HeaderToolbar`,
`UserMenu`,
`SystemMenu`,
`EnvironmentMenu`

Adds a test to `HeaderToolbar/utils.ts`

### CODE REVIEW

File: src/components/Layout/HeaderToolbar/utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+describe('HeaderToolbar utils', () => {
+  describe('parseEnvironments', () => {
+    it('parse WalkMeEnvironment[] to IWMDropdownOption[] correctly', () => {
+      const environments = [
+        { id: 0, name: 'env-1' },
+        { id: 1, name: 'env-2' },
+      ] as WalkMeEnvironment[];
+
+      const options = parseEnvironments(environments);
+      const valuesAreMatches = options.every((o, index) => o.value === environments[index].name);
```
Comment (dvira) [inline src/components/Layout/HeaderToolbar/utils.test.ts:16]: `valuesAreMatches` is grammatically incorrect. Consider changing to `valuseMatch` or `allValuesMatch` or  `areMatchingValues`.

File: src/components/Layout/HeaderToolbar/utils.test.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import { SystemData, WalkMeEnvironment } from '@walkme/editor-sdk';
+
+import { IWMDropdownOption } from '../../common/WMDropdown';
+
+import { parseEnvironments, getParsedEnvironment, parseSystems } from './utils';
+
+describe('HeaderToolbar utils', () => {
+  describe('parseEnvironments', () => {
+    it('parse WalkMeEnvironment[] to IWMDropdownOption[] correctly', () => {
```
Comment (dvira) [inline src/components/Layout/HeaderToolbar/utils.test.ts:9]: `should parse...` or `parses...`

File: src/components/Layout/HeaderToolbar/utils.test.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      ] as WalkMeEnvironment[];
+
+      const options = parseEnvironments(environments);
+      const valuesAreMatches = options.every((o, index) => o.value === environments[index].name);
+      expect(valuesAreMatches).toBeTruthy();
+    });
+  });
+
+  describe('getParsedEnvironment', () => {
+    it('return parsed environment as IWMDropdownOption correctly', () => {
```
Comment (dvira) [inline src/components/Layout/HeaderToolbar/utils.test.ts:22]: `should return...` or `returns..`

File: src/components/Layout/HeaderToolbar/utils.test.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    });
+  });
+
+  describe('parseSystems', () => {
+    const systems = [
+      { userId: 0, displayName: 'system-1' },
+      { userId: 1, displayName: 'system-2' },
+    ] as SystemData[];
+
+    it('return array', () => {
```
Comment (dvira) [inline src/components/Layout/HeaderToolbar/utils.test.ts:37]: Same grammar comment as before.

File: src/components/Layout/HeaderToolbar/utils.test.ts [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+      const options = parseSystems(systems) as IWMDropdownOption[];
+      expect(options).toHaveLength(systems.length);
+
+      const valuesAreMatches = options.every(
```
Comment (dvira) [inline src/components/Layout/HeaderToolbar/utils.test.ts:41]: See my previous suggestions.

File: src/components/Layout/HeaderToolbar/utils.test.ts [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+        (o, index) => o.id === systems[index].userId && o.value === systems[index].displayName,
+      );
+      expect(valuesAreMatches).toBeTruthy();
+    });
+
+    it('return object', () => {
```
Comment (dvira) [inline src/components/Layout/HeaderToolbar/utils.test.ts:47]: Same grammar comment as before.

File: src/components/common/dialogs/ImpersonateDialog/index.tsx [hunk 7]
```diff
@@ -67,7 +68,7 @@ export default function ImpersonateDialog({
       onConfirm={() => onConfirmHandle(localValue)}
       disableConfirmButton={invalid}
       confirmLabel="impersonate"
-      className={classes['impersonate-dialog']}
+      className={cc([classes['impersonate-dialog'], { open, close: !open }])}
```
Comment (dvira) [inline src/components/common/dialogs/ImpersonateDialog/index.tsx:71]: IMO, adding the 'close' class is redundant, since the 'open' should be enough, no?

**Review by dvira — APPROVED** [review]
Amazing job!
Left some suggestions to consider.

File: src/components/Layout/HeaderToolbar/HeaderToolbar.test.tsx [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+          ...initialState.originalUser,
+          userID: mockOriginalUserId,
+        },
+      },
+    ]),
+  };
+});
+
+describe('HeaderToolbar', () => {
+  it('should renders correctly', () => {
```
[the remark this answers is not in the document]
Comment (nunibaranes) [reply src/components/Layout/HeaderToolbar/HeaderToolbar.test.tsx:41]: 😅

File: src/components/Layout/HeaderToolbar/SystemMenu.test.tsx [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+    expect(systemMenu).toBeInTheDocument();
+
+    const options = screen.getAllByRole('menuitem');
+    expect(options).toHaveLength(defaultSystems.length);
+  });
+
+  it('change system', async () => {
+    const newSystem = defaultSystems[1];
+    const systemMenuBtn = renderSystemMenu();
+    await getSystemMenu();
```
[the remark this answers is not in the document]
Comment (nunibaranes) [reply src/components/Layout/HeaderToolbar/SystemMenu.test.tsx:86]: You right, thanks

File: src/components/Layout/HeaderToolbar/UserMenu.test.tsx [hunk 10]
```diff
@@ -1,14 +1,111 @@
...
+    const options = screen.getAllByRole('menuitem');
+    expect(options).toHaveLength(2);
+  });
+
+  it('click on Log out', async () => {
+    renderUserMenu();
+    await getUserMenu();
+    const logOutOption = screen.getByText('Log Out').closest('li');
+    logOutOption && userEvent.click(logOutOption);
+    expect(logout).toHaveBeenCalled();
```
[the remark this answers is not in the document]
Comment (nunibaranes) [reply src/components/Layout/HeaderToolbar/UserMenu.test.tsx:77]: Screen Shot 2020-10-12 at 16 20 10

If I don't mock the `logout` function I get the error above,
if I don't import the actual `logout` I get an error that logout is not defined.
also, my test fails.

File: src/components/common/WMDialog/index.tsx [hunk 11]
```diff
@@ -49,6 +49,7 @@ export default function WMDialog({
       maskStyle={{ backgroundColor: '#0B0B0B80' }}
       style={{ maxWidth: 392 }}
       closeIcon={<CloseButton />}
+      data-testid="wm-modal"
```
Comment (dvira) [inline src/components/common/WMDialog/index.tsx:52]: This is not used anywhere.

**Review by dvira — APPROVED** [review]

**Pull request merged**